udp_mux.go

```diff
@@ -16,6 +19,7 @@ type UDPMux interface {
 	io.Closer
 	GetConn(ufrag string) (net.PacketConn, error)
 	RemoveConnByUfrag(ufrag string)
+	GetXORMappedAddr(serverAddr net.Addr, deadline time.Duration) (*stun.XORMappedAddress, error)
```

Comment:
Adding `GetXORMappedAddr` breaks the public API.

Does this matter, does anyone implement? If everyone uses our `Default` I think it is ok.

Reply:
Good point and a good question :)  
The thing is that if we want to be universal and support all the types of candidates and not only host candidates then this interface should change.
Also, we will be adding **GetRelayedAddr** function in the next steps to support relay as well. 

One option we could choose not to break changes is to introduce a new interface **UniversalUDPMux** which will eventually replace the original **UDPMux** (which can be deprecated at this point).

I'm hoping that everyone uses the default UDPMux :) 

What are your thoughts on that @Sean-Der @davidzhao @boks1971 ?

Reply:
At least I see that it won't break things in pion/webrtc cuz the default one is used
https://github.com/pion/webrtc/blob/master/icemux.go#L22

Reply:
this probably should use a different interface. it doesn't feel that the simple UDPMux should be concerned with STUN or XORMappedAddr.

Reply:
Got you,

To keep muxing logic separate I think we could then add a layer (struct) that reads from the UDPConn and handles STUN packets before the muxer. But then, this struct has to always wrap the muxer and should be exposed to the agent user. 
I'm happy to jump in a call and discuss this further. We are very much looking forward to improving this part further and making pion/ice more efficient.

Reply:
Agreed with @davidzhao that we'll create a separate SRFLXUDPMux